### ISSUE TITLE
Develop roadmap

### ISSUE THREAD

**Issue by elight** [rubypair/rubypair#27]
We want users to come to the site, create a profile to specify and advertise their interests and location to the community, and indicate whether they want to pair remote or in-person.

Where do we go from here?

I've considered creating the following in an order something like this:
1. Recommendation engine based on user's location, pairing preferences, and interests to recommend users to pair with.  This incorporates several issues currently tracked (e.g., geolocation, spatial query, and recommendation engine based on user profile), Show recommendations to users when they log in.  Maybe occasionally email them with new recommendations (e.g., the system finds better matches for them than before)
2. Do more to facilitate communication within the application.  Maybe a commenting system, links to calendars (e.g., Google appoints, tungle.me).

What else would you find useful?

We should absolutely not try to do everything.  Better to leverage other people's apps than build every feature ourselves.

**Comment by lenary** [comment]
> We should absolutely not try to do everything. Better to leverage other people's apps than build every feature ourselves.

So i think now would be a good opportunity to spin up websites that can start making money and have good integration with rubypair - without being the same codebase.

I raised the idea of a SSH Tunnelling server and a pairing scheduler - both of which can be done as independent applications if this would keep rubypair simpler.

**Comment by pythonandchips** [comment]
Don't know if it is any help but I wrote a bunch of cucumber scenarios when I started to put together something similar earlier this year (unfortunately real life got in the way and didn't get much further that this).

https://raw.github.com/pythonandchips/pair-me-old/master/features/pair_me.feature

feel free to rip them off, ignore them or whatever.

**Comment by elight** [comment]
+1 (I should be sleeping)

**Comment by elight** [comment]
So I'm continuing to get more requests for location/distance-based searches for pairees.

## Next few weeks

Perhaps even before (1) above, we should geocode everyone that we can and then provide a spatal search for pairees.  That should be even simpler than a recommendation engine -- but provides some of the prerequisites for it.

Commenting is also low-hanging fruit.  Raz already has a pull request in for that should provide most/all of that.

And we have one or two nuisance bugs to clean up.

## Longer term

Let's get the above features done then see what people clamor for.  Folks have been fairly vocal about location-based searching.  After that, I can make some suggestions on Twitter via @rubypair and see what people say.

Or, hell, maybe we put a poll on http://rubypair.com?

Thoughts?

**Comment by lenary** [comment]
I don't think geocoding everyone's location will be _that_ hard, and then we can use geo searches and navigator.location to work out people close to them. (mongo should support that)

I think it might also be good to have large indexes of everyone - but not sure how to do that. perhaps this calls for a data migration :/

**Comment by deJaVisions** [comment]
Having a chat app, or maybe just an #rubypair irc channel elsewhere would be nice..so that you could talk to people in realtime.

**Comment by elight** [comment]
doktorj: We already have an unofficial #rubypair on freenode. ;-)

**Comment by elight** [comment]
lenary: a "data migration", in this case, is not necessarily anything more than a method(s) and/or rake task that may use said method.  Not too big a deal.  As we discussed on IRC, the migration could be run on one of our machines.  I'm fine with running it locally on mine.

It would just need to be TDD'd and tested against a local copy of the database.

I've already dumped the DB before.  As we're only up to 360 users, it's trivial to get a data dump.  Having simulated production data to test against is a simple matter.

Sounds like we're ready to proceed then.  I'll mark this issue closed as I've gone through and marked the related features as being on the 1.1 milestone.
